File: frontend/src/types/webhook.ts
export type WebhookEvent = 'lead.created' | 'lead.updated' | 'lead.status_changed'

export const WEBHOOK_EVENTS: WebhookEvent[] = [
  'lead.created',
  'lead.updated',
  'lead.status_changed',
]

export interface Webhook {
  id: string
  url: string
  secret: string
  events: WebhookEvent[]
  enabled: boolean
  description: string | null
  created_at: number
  updated_at: number
}

export interface WebhookDelivery {
  id: string
  webhook_id: string
  event: WebhookEvent
  payload: Record<string, unknown>
  status: 'pending' | 'delivered' | 'failed'
  attempt: number
  status_code: number | null
  error: string | null
  next_retry_at: number | null
  delivered_at: number | null
  created_at: number
}

export interface WebhookCreate {
  url: string
  events: WebhookEvent[]
  enabled?: boolean
  description?: string | null
}

export interface WebhookUpdate {
  url?: string
  events?: WebhookEvent[]
  enabled?: boolean
  description?: string | null
}

File: backend/app/schemas/webhook.py
from pydantic import BaseModel, Field, HttpUrl

VALID_EVENTS = {"lead.created", "lead.updated", "lead.status_changed", "signal.detected", "lead.hot"}


class WebhookCreate(BaseModel):
    url: HttpUrl
    events: list[str] = Field(default_factory=lambda: list(VALID_EVENTS))
    enabled: bool = True
    description: str | None = None


class WebhookUpdate(BaseModel):
    url: HttpUrl | None = None
    events: list[str] | None = None
    enabled: bool | None = None
    description: str | None = None


class WebhookRead(BaseModel):
    id: str
    url: str
    secret: str
    events: list[str]
    enabled: bool
    description: str | None
    created_at: int
    updated_at: int

    model_config = {"from_attributes": True}


class WebhookDeliveryRead(BaseModel):
    id: str
    webhook_id: str
    event: str
    payload: dict
    status: str
    attempt: int
    status_code: int | None
    error: str | None
    next_retry_at: int | None
    delivered_at: int | None
    created_at: int

    model_config = {"from_attributes": True}
